base_flexcons.py

```diff
@@ -0,0 +1,325 @@
+
+    @abstractmethod
+    def adjust_threshold(self, local_measure):
+        """
+        Método abstrato para ajuste do threshold.
+        Cada classe derivada deve implementar sua lógica específica.
+        """
+        pass
+
+    def fit(self, X, y):
+        """
+        Treina o classificador usando X e y como dados de treinamento.
+
+        Args:
+            X (array-like, shape = [n_samples, n_features]):
+                Matriz de características contendo os dados para o treinamento.
+            y (array-like, shape = [n_samples]):
+                Vetor de rótulos associados aos dados de entrada.
+                Valores de rótulo não rotulados devem ser representados por -1.
+
+        Returns:
+            Object: Classificador treinado
+        """
+        # Inicialização do classificador e acurácia inicial
+        labeled_indices = np.where(y != -1)[0]
+        unlabeled_indices = np.where(y == -1)[0]
+
+        self.train_new_classifier(labeled_indices, X, y)
+
+        for self.n_iter_ in range(self.max_iter):
+
+            # Verifica se ainda há instâncias não rotuladas
+            if len(unlabeled_indices) == 0:
+                break
+
+            # Fazer previsões e selecionar instâncias
+            self.pred_x_it = self.storage_predict(
+                unlabeled_indices,
+                self.classifier_.predict_proba(X[unlabeled_indices]).max(axis=1),
+                self.classifier_.predict(X[unlabeled_indices])
+            )
+            selected_indices, predictions = self.select_instances_by_rules()
+
+            if len(selected_indices) == 0:
+                break
+
+            # Atualizar o conjunto de instâncias rotuladas
+            self.add_new_labeled(selected_indices,selected_indices,predictions)
+
+            # Calcula a métrica local (usada para ajuste do threshold)
+            local_measure = self.calc_local_measure(
+                X[safe_mask(X, labeled_indices)],
+                y[labeled_indices],
+                self.classifier_
+            )
+
+            # Atualiza o histórico de precisão
+            self.accuracy_history.append(local_measure)
+
+            # Ajuste do threshold baseado em critério específico da subclasse
+            self.adjust_threshold(local_measure)
+
+            # Re-treinar o classificador com o novo conjunto de dados rotulados
+            self.train_new_classifier(labeled_indices, X, y)
+
+        return self
+
+    def calc_local_measure(self, X, y_true, classifier):
+        """
+        Calcula o valor da acurácia do modelo
+
+        Args:
+            X: instâncias
+            y_true: classes
+            classifier: modelo
+
+        Returns:
+            Retorna a acurácia do modelo
+        """
+        y_pred = classifier.predict(X)
+        return accuracy_score(y_true, y_pred)
+
+    def update_memory(
+        self,
+        instances: List,
+        labels: List,
+        weights: Optional[List] = None
+    ):
```

Comment:
```suggestion
    def update_memory(
        self,
        instances: list[tuple],
        labels: list[int],
        weights: list[float] | None = None
    ):
```
- list[float] | None: está dizendo que pode ser uma lista ou None, o normal é ser None.